build_tools/benchmarks/post_benchmark_comment.py

```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/env python3
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+"""
+Posts benchmark results to gist and comments on pull requests.
+
+Requires the environment variables:
+
+- GITHUB_TOKEN: token from GitHub action that has write access on issues. See
+  https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token
+- GIST_BOT_USER: user name that posts the gist.
+- GIST_BOT_TOKEN: token that has write access to gist. Gist will be posted as
+  the owner of the token. See
+  https://docs.github.com/en/rest/overview/permissions-required-for-fine-grained-personal-access-tokens#gists
+"""
+
+import sys
+import pathlib
+
+# Add build_tools python dir to the search path.
+sys.path.insert(0, str(pathlib.Path(__file__).parent.with_name("python")))
+
+from typing import Any, Optional
+import argparse
+import http.client
+import json
+import os
+import requests
+
+from reporting import benchmark_comment
+
+GITHUB_IREE_API_PREFIX = "https://api.github.com/repos/iree-org/iree"
+GITHUB_GIST_API = "https://api.github.com/gists"
+GITHUB_API_VERSION = "2022-11-28"
+MAX_PAGES_TO_SEARCH_PREVIOUS_COMMENT = 10
+
+
+class APIRequester(object):
+  """REST API client that injects proper GitHub authentication headers."""
+
+  def __init__(self, github_token: str):
+    self._api_headers = {
+        "Accept": "application/vnd.github+json",
+        "Authorization": f"token {github_token}",
+        "X-GitHub-Api-Version": GITHUB_API_VERSION,
+    }
+
+  def get(self, endpoint: str, payload: Any) -> requests.Response:
+    return requests.get(endpoint,
```

Comment:
Use a session to reuse the connection? It can be instantiated with the object

Reply:
Done.